crud/stats/module.lua

```diff
@@ -0,0 +1,256 @@
+---- CRUD statistics module
+-- @module crud.stats.module
+--
+
+local clock = require('clock')
+local checks = require('checks')
+local errors = require('errors')
+local vshard = require('vshard')
+
+local dev_checks = require('crud.common.dev_checks')
+local utils = require('crud.common.utils')
+local op_module = require('crud.stats.operation')
+local registry = require('crud.stats.local_registry')
+local stash = require('crud.stats.stash')
+
+local StatsError = errors.new_class('StatsError', {capture_stack = false})
+
+local stats = {}
+local internal = stash.get('internal')
+
+--- Check if statistics module was enabled.
+--
+-- @function is_enabled
+--
+-- @treturn boolean Returns `true` or `false`.
+--
+function stats.is_enabled()
+    return internal.is_enabled == true
+end
+
+--- Initializes statistics registry, enables callbacks and wrappers.
+--
+--  If already enabled, do nothing.
+--
+-- @function enable
+--
+-- @treturn boolean Returns `true`.
+--
+function stats.enable()
+    if stats.is_enabled() then
+        return true
+    end
+
+    StatsError:assert(
+        rawget(_G, 'crud') ~= nil,
+        "Can be enabled only on crud router"
+    )
+
+    internal.is_enabled = true
+    registry.init()
+
+    return true
+end
```

Comment:
Probably it could be simplifed to

```lua
if need_to_enable then
    require('crud').insert = require('crud.insert').call
else
    require('crud').insert = wrap(require('crud.insert').call)
end
```

I see that it's a way how to achieve "zero-overhead" in case if metrics disabled and drop some if-checks on hot paths.

Reply:
I don't see a way to make a convenient handle for a user to enable/disable metrics with this approach.

And I want to highlight that performance tests shows that there is no distinctive difference between using `(...)` wrapper and not using any wrappers for CRUD calls.

Reply:
I would not introduce the extra complexity if performance measurements do not show a difference.

Reply:
I'll consider it as resolved.